games-emulation/ppsspp/ppsspp-1.11.3_p20210627.ebuild

```
@@ -0,0 +1,118 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake desktop xdg
+
+DESCRIPTION="A PSP emulator written in C++"
+HOMEPAGE="https://www.ppsspp.org/"
+COMMIT="ff2a6eeda2d00210f1a1f7bc0ab0ad99e7bc60b6"
+SRC_URI="
+	https://github.com/hrydgard/${PN}/archive/${COMMIT}.tar.gz -> ${P}.tar.gz
+	https://github.com/KhronosGroup/SPIRV-Cross/archive/9acb9ec31f5a8ef80ea6b994bb77be787b08d3d1.tar.gz -> ${P}-ext_SPIRV-Cross.tar.gz
+	https://github.com/Kingcom/armips/archive/7885552b208493a6a0f21663770c446c3ba65576.tar.gz -> ${P}-ext_armips.tar.gz
+	https://github.com/Tencent/rapidjson/archive/73063f5002612c6bf64fe24f851cd5cc0d83eef9.tar.gz -> ${P}-ext_rapidjson.tar.gz
+	https://github.com/facebook/zstd/archive/56421f34e4ab5c17cbac0832bd12413e4f27ecf9.tar.gz -> ${P}-ext_zstd.tar.gz
+	https://github.com/hrydgard/glslang/archive/dc11adde23c455a24e13dd54de9b4ede8bdd7db8.tar.gz -> ${P}-ext_glslang.tar.gz
+	https://github.com/hrydgard/miniupnp/archive/3a87be33e797ba947b2b2a5f8d087f6c3ff4d93e.tar.gz -> ${P}-ext_miniupnp.tar.gz
+	https://github.com/hrydgard/ppsspp-freetype/archive/cbea79dc8fef4d9210e2bac7e7b9b5ff3388197a.tar.gz -> ${P}-ext_native_tools_prebuilt.tar.gz
+	https://github.com/hrydgard/ppsspp-lang/archive/3bf6b18eb0ae68861fbfbb7b43cd764771ca42a9.tar.gz -> ${P}-assets_lang.tar.gz
+	!system-ffmpeg? ( https://github.com/hrydgard/ppsspp-ffmpeg/archive/0b28335acea4f429ae798c5e75232e54881bf164.tar.gz -> ${P}-ffmpeg.tar.gz )
+	discord? ( https://github.com/discordapp/discord-rpc/archive/963aa9f3e5ce81a4682c6ca3d136cddda614db33.tar.gz -> ${P}-ext_discord-rpc.tar.gz )
+"
+
+RESTRICT="test"
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="discord qt5 sdl +system-ffmpeg"
+REQUIRED_USE="!qt5? ( sdl )"
+
+RDEPEND="
+	app-arch/snappy:=
+	dev-libs/libzip:=
+	dev-util/glslang:=
+	media-libs/glew:=
+	sys-libs/zlib:=
+	virtual/opengl
+	sdl? ( media-libs/libsdl2 )
+	system-ffmpeg? ( media-video/ffmpeg:= )
+	qt5? (
+		dev-qt/qtcore:5
+		dev-qt/qtgui:5[-gles2-only]
+		dev-qt/qtmultimedia:5[-gles2-only]
+		dev-qt/qtopengl:5[-gles2-only]
+		dev-qt/qtwidgets:5[-gles2-only]
+	)
+"
+DEPEND="${RDEPEND}"
+
+S="${WORKDIR}/${PN}-${COMMIT}"
+
+src_unpack() {
+	unpack "${P}.tar.gz"
+
+	cd "${S}"  || die
+	local list=(
+		assets_lang
+		ext_SPIRV-Cross
+		ext_armips
+		ext_glslang
+		ext_miniupnp
+		ext_native_tools_prebuilt
+		ext_rapidjson
+		ext_zstd
+	)
+	! use system-ffmpeg  && list+=( ffmpeg )
+	use discord  && list+=( ext_discord-rpc )
+
+	local i
+	for i in "${list[@]}"; do
+		tar xf "${DISTDIR}/${P}-${i}.tar.gz" --strip-components 1 -C "${i//_//}"  ||
+			die "Failed to unpack ${P}-${i}.tar.gz"
+	done
+}
+
+src_prepare() {
+	if ! use system-ffmpeg; then
+		sed -i -e "s#-O3#-O2#g;" "${S}"/ffmpeg/linux_*.sh  || die
+	fi
+
+	cmake_src_prepare
+}
+
+src_configure() {
+	local mycmakeargs=(
+		$(cmake_use_find_package sdl SDL2)
```

Note that if going with removing IUSE=sdl, can do this:
```suggestion
		$(cmake_use_find_package !qt5 SDL2)
```

---

Although if need to check for joystick too, I guess this will need to be added separately now.

I do admit it'd be simpler to be an unconditional dep, then could remove this, but well. I'll let you decide. I do feel like not many would want this without joystick support.

---

> I do feel like not many would want this without joystick support.

Yea, I definitely want it with :D